### ISSUE TITLE
Abort due to constraint violation -- this is not the repaint issue other users were reporting

### ISSUE THREAD

**Issue by tssajo** [toggl/toggldesktop#1706]
Hello,

Since my Toggl Desktop upgraded itself a few weeks ago (I'm on the stable channel, btw) I constantly run into this issue:
![image](https://cloud.githubusercontent.com/assets/2164104/11500125/61cb753a-982b-11e5-8dcf-46f837b7bf26.png)
Once this happens, I can no longer do anything: None of the buttons work, for example I cannot even stop the timer by clicking on the Stop button, the mouse click triggers nothing. :(

The version which I have now is 7.3.157. I am on Windows 7 (64bit). I have a 3-monitor setup and I run Toggl Desktop on my 2nd monitor which is on the right side. My video card is an onboard Intel HD Graphics 4600. Toggl Desktop worked perfectly on this very same PC before. Previous versions such as 7.2.266 and 7.2.281 worked just fine. I believe this problem started to appear since version 7.3.xxx got released and pushed to my PC. It is too bad automatic updates cannot be disabled therefore I cannot downgrade to a known good version of Toggl Desktop. :(

The only "workaround" is this: When this happens, I must completely exit from the Toggl Desktop program and re-launch it a few seconds later. Then it works again just fine -- for a while. This is extremely annoying since this "Abort due to constraint violation" error occurs after less than a minute of the (re-)launching of the program! This means that I would need to keep quitting and re-launching Toggle Desktop all day long... :(

Please advise. Thanks!

**Comment by amulware** [comment]
Thanks for reporting the problem!

There may be some corrupt data in the local database.
Have you tried clearing the app's cache? (In the cog-wheel context menu)

Let us know if that helps!

**Comment by tssajo** [comment]
@amulware I just cleaned the cache, it logged me out right away. When I tried to log back in I immediately got another "Abort due to constraint violation" ... :( So I could not even log back in.

**Comment by tssajo** [comment]
@amulware Man, since I cleaned the cache, I can no longer log back in to one of the accounts I use. I have two accounts and I can login to my free account with Toggl Desktop. But I also got invited to a paid account, actually the person I'm doing work for is paying for that Toggl account. I can no longer log in to that account with Toggl Desktop! But of course I can login via the web interface using a web browser. This is really bad! Can you help me with this? Thanks!

**Comment by IndrekV** [comment]
@tssajo Please write to our support support@toggl.com with details (email used for the account) and we'll get this sorted.

**Comment by tssajo** [comment]
@IndrekV Okay, I sent a message to support@toggl.com about two hours ago but there is no response yet. Is it that hard to get it fixed? I cannot use Toggl Desktop at all now, and it is a paid account, you know. Could you please ask someone to please look into it? Thanks!

**Comment by tssajo** [comment]
Thanks to IndrekV's email assistance, I was able to fix this issue by doing the following:

I opened up Toggl Desktop and cleared the cache (from under the cog wheel), then I logged out and exited from the program. Then I went to the Control Panel of my PC and uninstalled Toggl Desktop from there. Then I deleted the following three directories (note: there were lots of sub directories and files under these) :

C:\Program Files (x86)\Toggl
C:\Users\<username>\AppData\Local\Toggl
C:\Users\<username>\AppData\Local\TogglDesktop

I don't know why Toggl Desktop created (over the years) so many directories on my computer... I also don't know what sort of garbage got collected under these directories. But I am glad to report that after deleting all these garbage (and after a PC reboot) and then reinstalling Toggl Desktop v7.3.157 my original issue is fixed! I hope this post can help someone with a similar problem.

Thanks!
